tree/treeplayer/src/TTreeFormula.cxx

```diff
@@ -2961,8 +2961,10 @@ Int_t TTreeFormula::DefinedVariable(TString &name, Int_t &action)
             if (current[0] == ']') {
                fIndexes[code][dim] = -1; // Loop over all elements;
             } else {
+               TString tempIndex(current);
+               tempIndex.Remove(TString::kTrailing, ' ').Remove(TString::kTrailing, ']');
                scanindex = sscanf(current,"%d",&index);
-               if (scanindex) {
+               if (scanindex == 1 && tempIndex.IsDigit()) {
```

Comment:
The 2 tests should be redundant (i.e. `scanindex == 1` implies at that point that ` tempIndex.IsDigit()` is true.

Reply:
Not really, according to my debugging, sscanf of 2-1 returns scanindex ==1. Otherwise the test case is not fixed without this line

Reply:
humm ... okay but why not put it 'before' the `sscanf` like the other updated code?

Reply:
I could do the nested if here too but then it's more involved since there is an if else structure in this case.
In any case, some tests are failing :s

Reply:
Both things fixed now